### PR TITLE
#23627 improved getCustomerName method for Order

### PR DESCRIPTION
**Pull request by sergiy-v**
### Description (*)
Improved `getCustomerName` method for `Order`.

### Fixed Issues (if relevant)
#23627 

### Manual testing scenarios (*)
- Register a new customer with prefix, first name, middle name, last name, and suffix.
- Place an order with the customer, check the customer name for the order confirmation email and customer details on order page, the prefix, middle name and suffix should be visible.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Sales/Model/Order.php [hunk 1]
```diff
@@ -1965,16 +1976,71 @@ public function getRelatedObjects()
      *
      * @return string
      */
-    public function getCustomerName()
+    public function getCustomerName(): string
```
Comment (dmytro-ch) [inline app/code/Magento/Sales/Model/Order.php]: Unfortunately, we are not allowed to change the signature of public and protected methods from the backward compatibility perspective. Please refer to the *Semantic Version Checker* report for more information.

File: app/code/Magento/Sales/Model/Order.php [hunk 2]
```diff
@@ -1965,16 +1976,71 @@ public function getRelatedObjects()
...
+
         return $customerName;
     }
 
+    /**
+     * Is visible customer middlename
+     *
+     * @return bool
+     */
+    private function isVisibleCustomerMiddlename(): bool
```
Comment (dmytro-ch) [inline app/code/Magento/Sales/Model/Order.php]: Could you please move the new private methods to the end of the class? So the public and protected methods would be followed by private methods.

**Review by dmytro-ch — CHANGES_REQUESTED** [review]
Hi @sergiy-v, thank you for your contribution! :)
Could you please fix _Semantic Version Checker_ and _Static Test_ failures?

File: app/code/Magento/Sales/Model/Order.php (hunk 2 above)
Comment (sergiy-v) [reply app/code/Magento/Sales/Model/Order.php]: Moved, thank you.

File: app/code/Magento/Sales/Model/Order.php (hunk 1 above)
Comment (sergiy-v) [reply app/code/Magento/Sales/Model/Order.php]: Thanks for the clarification.

**Review by dmytro-ch — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @dmytro-ch, thank you for the review.
[ENGCOM-6039](https://jira.corp.magento.com/browse/ENGCOM-6039) has been created to process this Pull Request
:eight_spoked_asterisk: @dmytro-ch, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA passed

**Comment by plastikschnitzer** [comment]
Is this fixed in Magento 2.3.5-p1? For me its not working there, middle name is showing up in the customer administration only, not in the order overview, order detail page or the transaction email.

**Pull request merged**